### PR TITLE
Change the name of Static Check without pylint

### PR DESCRIPTION
**Pull request by kaxil**
This name was confusing, hopefully it is better now

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by kaxil** [comment]
Let me know if this is better, if not we could close this PR

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -155,7 +155,7 @@ jobs:
 
   static-checks:
     timeout-minutes: 30
-    name: "Static checks: no pylint"
+    name: "Static checks: except pylint"
```
Comment (turbaszek) [inline .github/workflows/ci.yml]: I think it's better. But maybe we should rename the other step to simply `Pylint` and name this one as `Static checks` in this way there will be no wording problem I think.

Comment (potiuk) [reply .github/workflows/ci.yml]: Just one watch out: https://github.com/apache/airflow/blob/master/.github/workflows/build-images-workflow-run.yml#L102. I think "Pylint" and "Static checks" is good but we have to update the cancelling workflow to make sure the new name is matched by the regexp :). No harm if it's not - but if it matches, then a failure in this step will trigger cancelling the whole workflow when someone starts another PR.

Comment (kaxil) [reply .github/workflows/ci.yml]: Updated

**Review by turbaszek — APPROVED** [review]

File: .github/workflows/build-images-workflow-run.yml [hunk 2]
```diff
@@ -99,7 +99,7 @@ jobs:
           cancelMode: failedJobs
           sourceRunId: ${{ github.event.workflow_run.id }}
           jobNameRegexps: >
-            ["^Static checks.*", "^Build docs$", "^Spell check docs$", "^Backport packages$",
+            ["^Pylint$", "^Static checks$", "^Build docs$", "^Spell check docs$", "^Backport packages$",
```
Comment (potiuk) [inline .github/workflows/build-images-workflow-run.yml:102]: There is a mistake here. We need to add the whole name of the check if you removed .*  as the regexp matches $ - end of line. It is "Static checks: except pylint" now.

Comment (kaxil) [reply .github/workflows/build-images-workflow-run.yml:102]: Oh wait, the job name is just "Static checks" the step name is "Static checks except pylint"

Comment (kaxil) [reply .github/workflows/build-images-workflow-run.yml:102]: So should be ok, no?

**Pull request merged**